### ISSUE TITLE
How to generate CSV to train on a new dataset

### ISSUE THREAD

**Issue by mario-m-g** [TaoRuijie/TalkNet_ASD#71]
Hello,

I am trying to train TalkNet on a different dataset but I don't understand how both CSV train_orig and train_loader work.
I have already seen https://github.com/TaoRuijie/TalkNet-ASD/issues/30  and  https://github.com/TaoRuijie/TalkNet-ASD/issues/10. But I still don't get what some of the fields mean. 
1. What do entity_id, label_id and instance_id mean?
2. A line in train_loader is, for instance: 8aMv-ZGD4ic_1500_1560:21	87	25.0.   I guess 25 is fps an 87 is the number of frames in that clip. And 21, what does that represent?
3. How is train_loader generated from train_orig? Is there any code for that? Is it created somewhere in the provided code and I have not seen it?

Thanks in advance!

**Comment by King-king424** [comment]
hey,  I want to know the relationship between entity_id and instance_id.
Have you figured it out, please！

**Comment by mario-m-g** [comment]
> hey, I want to know the relationship between entity_id and instance_id. Have you figured it out, please！

Hi, sorry I closed this without leaving any comments. Here's what I found out:

Entity_id is just a person ID. It always starts from 1 (there's no person 0) and is just used to link the bounding boxes and labels with a specific person. When there is a scene change, entity_id starts from 1 again, even if it is the same person as in the previous scene.

Instance_id is a counter that increases by one every time a person switches from 'SPEAKING_AUDIBLE' to 'NOT_SPEAKING' or vice versa. This one starts from 0 and is linked to a person,  which is why it's built by concatenating the entity_id with the counter.

Anyway, I found out that instance_id is never used in the code (I think), so I wouldn't worry about it. Check the preprocess_AVA function, which is the only one that uses the train/val/test_orig files.

Hope this helps. Let me know if you need anything else!

**Comment by King-king424** [comment]
> > hey, I want to know the relationship between entity_id and instance_id. Have you figured it out, please！
> 
> Hi, sorry I closed this without leaving any comments. Here's what I found out:
> 
> Entity_id is just a person ID. It always starts from 1 (there's no person 0) and is just used to link the bounding boxes and labels with a specific person. When there is a scene change, entity_id starts from 1 again, even if it is the same person as in the previous scene.
> 
> Instance_id is a counter that increases by one every time a person switches from 'SPEAKING_AUDIBLE' to 'NOT_SPEAKING' or vice versa. This one starts from 0 and is linked to a person, which is why it's built by concatenating the entity_id with the counter.
> 
> Anyway, I found out that instance_id is never used in the code (I think), so I wouldn't worry about it. Check the preprocess_AVA function, which is the only one that uses the train/val/test_orig files.
> 
> Hope this helps. Let me know if you need anything else!

Thanks，I've already understood. Your explanation is very clear.
I have other question. I'm now trying to construct my own dataset for training. I want to know about face clips and audio clips.
Will the target face crops always exist during the duration of an audio clip?

**Comment by mario-m-g** [comment]
> > > hey, I want to know the relationship between entity_id and instance_id. Have you figured it out, please！
> > 
> > 
> > Hi, sorry I closed this without leaving any comments. Here's what I found out:
> > Entity_id is just a person ID. It always starts from 1 (there's no person 0) and is just used to link the bounding boxes and labels with a specific person. When there is a scene change, entity_id starts from 1 again, even if it is the same person as in the previous scene.
> > Instance_id is a counter that increases by one every time a person switches from 'SPEAKING_AUDIBLE' to 'NOT_SPEAKING' or vice versa. This one starts from 0 and is linked to a person, which is why it's built by concatenating the entity_id with the counter.
> > Anyway, I found out that instance_id is never used in the code (I think), so I wouldn't worry about it. Check the preprocess_AVA function, which is the only one that uses the train/val/test_orig files.
> > Hope this helps. Let me know if you need anything else!
> 
> Thanks，I've already understood. Your explanation is very clear. I have other question. I'm now trying to construct my own dataset for training. I want to know about face clips and audio clips. Will the target face crops always exist during the duration of an audio clip?

Face crops are generated using the bounding boxes from the CSV file. If there is no bounding box in a specific frame of the clip, it doesn't affect the audio clip. I think audio clips are the same for each person in the scene. I'm not sure if this is what you meant.
